core/src/main/java/org/polypheny/db/catalog/snapshot/impl/SnapshotImpl.java

```diff
@@ -0,0 +1,333 @@
+        return adapterTemplates.values().stream().filter( t -> t.adapterName.equalsIgnoreCase( name ) && t.adapterType == adapterType ).findAny();
+    }
+
+
+    @Override
+    public List<AdapterTemplate> getAdapterTemplates( AdapterType adapterType ) {
+        return adapterTemplates.values().stream().filter( t -> t.adapterType == adapterType ).collect( Collectors.toList() );
+    }
+
+
+    @Override
+    public List<QueryInterfaceTemplate> getInterfaceTemplates() {
+        return List.copyOf( interfaceTemplates.values() );
+    }
+
+
+    @Override
+    public @NotNull Optional<LogicalEntity> getLogicalEntity( long namespaceId, String entity ) {
+        Optional<LogicalTable> optionalTable = rel().getTable( namespaceId, entity );
+        if ( optionalTable.isPresent() ) {
+            return Optional.of( optionalTable.get() );
+        }
+        Optional<LogicalCollection> optionalCollection = doc().getCollection( namespaceId, entity );
+        if ( optionalCollection.isPresent() ) {
+            return Optional.of( optionalCollection.get() );
+        }
+        Optional<LogicalGraph> optionalGraph = graph().getGraph( namespaceId );
+        if ( optionalGraph.isPresent() ) {
+            return Optional.of( optionalGraph.get() );
+        }
+        return Optional.empty();
+    }
+
+
+    @Override
+    public boolean equals( Object o ) {
+        if ( this == o ) {
+            return true;
+        }
+        if ( o == null || getClass() != o.getClass() ) {
+            return false;
+        }
+
+        SnapshotImpl snapshot = (SnapshotImpl) o;
+
+        if ( !rel.equals( snapshot.rel ) ) {
+            return false;
+        }
+        if ( !doc.equals( snapshot.doc ) ) {
+            return false;
+        }
+        if ( !graph.equals( snapshot.graph ) ) {
+            return false;
+        }
+        if ( !alloc.equals( snapshot.alloc ) ) {
+            return false;
+        }
+        if ( !users.equals( snapshot.users ) ) {
+            return false;
+        }
+        if ( !userNames.equals( snapshot.userNames ) ) {
+            return false;
+        }
+        if ( !interfaces.equals( snapshot.interfaces ) ) {
+            return false;
+        }
+        if ( !interfaceTemplates.equals( snapshot.interfaceTemplates ) ) {
+            return false;
+        }
+        if ( !interfaceNames.equals( snapshot.interfaceNames ) ) {
+            return false;
+        }
+        if ( !adapters.equals( snapshot.adapters ) ) {
+            return false;
+        }
+        if ( !adapterTemplates.equals( snapshot.adapterTemplates ) ) {
+            return false;
+        }
+        if ( !adapterNames.equals( snapshot.adapterNames ) ) {
+            return false;
+        }
+        if ( !namespaces.equals( snapshot.namespaces ) ) {
+            return false;
+        }
+        return namespaceNames.equals( snapshot.namespaceNames );
+    }
+
+
+    @Override
+    public int hashCode() {
```

Comment:
Why not use lombok for equals and hashCode. Please document.

Reply:
Artifact, removed.